### PR TITLE
Fixing yaml dependency installation

### PR DESCRIPTION
**Pull request by Hazhzeng**
### The dependency installation procedure should only work when a dependency description file is presented in the local repository.

**This pull request is intended to fix the issues in following scenarios:**
1. Failure when a python repository **does not have a requirements.txt** file. The build should not fail and the "pip install" step should be skipped.
2. Failure when a javascript repository **does not have a package.json** file. The build should not fail and the "npm install" step should be skipped.
3. Failure when a javascript repository with a package.json file, but it does not have a "build" command defined in the "scripts" block. The build should not fail and the "npm build" step should be skipped.

### Miscellaneous
1. Currently we do not support Java and Powershell in devops-build tool. We should remove them from local repository.

### CODE REVIEW

**Comment by Hazhzeng** [comment]
Hey @maiqbal11, could you take a look in this PR? It is a quick fix for YAML generation (when dependency file is missing). Feel free the leave comments. Thanks for your time.

**Comment by Hazhzeng** [comment]
cc @Francisco-Gamino

File: azure_functions_devops_build/yaml/yaml_manager.py [hunk 1]
```diff
@@ -39,14 +41,8 @@ def create_yaml(self):
             language_str = 'dotnet'
             package_route = '$(System.DefaultWorkingDirectory)/publish_output/s'
             dependencies = self._dotnet_dependencies()
-        elif self._language == JAVA:
```
Comment (maiqbal11) [inline azure_functions_devops_build/yaml/yaml_manager.py]: Can we move the removing Java support into a separate PR? It will keep this PR focused on removing the dependencies path. We can also easily revert that change once we do end up supporting Java. Right now it would be difficult to easily undo the "disabling Java" part of this PR since you would have to go in and manually add back the removed code (which is prone to error).

File: azure_functions_devops_build/yaml/yaml_manager.py [hunk 2]
```diff
@@ -113,13 +132,3 @@ def _dotnet_dependencies(self):
         dependencies.append("    modifyOutputPath: true")
         dependencies.append("    zipAfterPublish: false")
         return dependencies
-
-    def _java_dependencies(self):
```
Comment (maiqbal11) [inline azure_functions_devops_build/yaml/yaml_manager.py]: Best to have this in a separate PR (rationale above).

File: azure_functions_devops_build/yaml/yaml_manager.py [hunk 3]
```diff
@@ -66,7 +62,24 @@ def _generate_yaml(self, dependencies, vmImage, language_str, platform_str, pack
...
+
+    def _requires_npm(self):
+        return path.exists('package.json')
+
+    def _requires_npm_build(self):
+        if path.exists('package.json'):
+            with open('package.json', 'r') as f:
+                json_object = json.load(f)
+                return bool(json_object.get('scripts', {}).get('build'))
+            return False
```
Comment (maiqbal11) [inline azure_functions_devops_build/yaml/yaml_manager.py]: Shouldn't need this line. The bool statement above should return True or False depending on what is found.

File: azure_functions_devops_build/yaml/yaml_manager.py [hunk 4]
```diff
@@ -66,7 +62,24 @@ def _generate_yaml(self, dependencies, vmImage, language_str, platform_str, pack
...
+        return False
+
+    def _requires_mvn(self):
```
Comment (maiqbal11) [inline azure_functions_devops_build/yaml/yaml_manager.py]: Do we need to add a Maven requirement here if we are disabling Java support?

File: azure_functions_devops_build/yaml/yaml_manager.py (hunk 1 above)
Comment (Hazhzeng) [reply azure_functions_devops_build/yaml/yaml_manager.py]: @maiqbal11 Good point! Reverted the Java support code. Will disable Java in a separate PR.

File: azure_functions_devops_build/yaml/yaml_manager.py (hunk 2 above)
Comment (Hazhzeng) [reply azure_functions_devops_build/yaml/yaml_manager.py]: Thanks, same as above. Reverted.

File: azure_functions_devops_build/yaml/yaml_manager.py (hunk 3 above)
Comment (Hazhzeng) [reply azure_functions_devops_build/yaml/yaml_manager.py]: Removed return statement.

File: azure_functions_devops_build/yaml/yaml_manager.py (hunk 4 above)
Comment (Hazhzeng) [reply azure_functions_devops_build/yaml/yaml_manager.py]: Reverted Maven change.

**Comment by Hazhzeng** [comment]
Thanks a lot @maiqbal11. Those are mistakes needed to be pointed out. Much appreciated 👍
